src/garage/sampler/ray_sampler.py

```diff
@@ -155,91 +160,30 @@ def _process_trajectory(self, result):
 class SamplerWorker:
     """Constructs a single sampler worker.
 
-    The worker can have its parameters updated, and sampler its policy for
-    trajectories or rollouts.
-
     Args:
-        - worker_id(int): the id of the sampler_worker
-        - env_pkl: A pickled gym or akro env object
-        - agent_pkl: A pickled agent
-        - max_path_length(int): max trajectory length
-        - should_render(bool): if true, renders trajectories after
-            sampling them
+        - worker_id(int): The id of the sampler_worker
+        - env_pkl: The pickled gym env
+        - agent_pkl: The pickled agent
+        - config(SamplerConfig): The sampler configuration
 
     """
 
-    def __init__(self,
-                 worker_id,
-                 env_pkl,
-                 agent_pkl,
-                 seed,
-                 max_path_length,
-                 should_render=False):
+    def __init__(self, worker_id, env_pkl, agent_pkl, config):
+        config.worker_init_fn(worker_id)
         self.worker_id = worker_id
-        self._env = pickle.loads(env_pkl)
+        self.env = pickle.loads(env_pkl)
         self.agent = pickle.loads(agent_pkl)
-        self._seed = seed + self.worker_id
-        deterministic.set_seed(self._seed)
-        self._max_path_length = max_path_length
-        self._should_render = should_render
-        self.agent_updates = 0
-
-    def set_agent(self, flattened_params):
-        """Set the agent params.
-
-        Args:
-            - flattened_params(): model parameters in numpy format
-        """
-        self.agent.set_param_values(flattened_params)
-        self.agent_updates += 1
+        self.config = config
+
+    def update(self, agent_update, env_update):
```

Comment:
@avnishn This function describes fair succinctly the relationship between the sampler, `config`, `agent_update_fn`, `env_update_fn`, `agent`, `agent_update`, `env`, and `env_update`.